### PR TITLE
Add option to override Kafka Broker version

### PR DESCRIPTION
**Pull request by krutsko**
# Description

Add option to override Kafka Broker version

## Issue reference

https://github.com/dapr/dapr/issues/3806

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation https://github.com/dapr/docs/issues/1912

### CODE REVIEW

File: pubsub/kafka/kafka_test.go [hunk 1]
```diff
@@ -40,16 +40,49 @@ func getBaseMetadata() pubsub.Metadata {
...
+func TestParseMetadata(t *testing.T) {
 	k := getKafkaPubsub()
-	meta, err := k.getKafkaMetadata(m)
-	assert.Nil(t, err)
+	t.Run("default kafka version", func(t *testing.T) {
+		m := getCompleteMetadata()
+		meta, err := k.getKafkaMetadata(m)
+		require.NoError(t, err)
+		assert.NotNil(t, meta)
+		assertMetadata(t, meta)
```
Comment (pkedy) [inline pubsub/kafka/kafka_test.go:60]: Nit: `assertMetadata` could be called once outside of the nested tests

File: pubsub/kafka/kafka_test.go [hunk 2]
```diff
@@ -40,16 +40,49 @@ func getBaseMetadata() pubsub.Metadata {
...
+		assert.Equal(t, sarama.V0_10_2_0, meta.Version)
+	})
+
+	t.Run("invalid kafka version", func(t *testing.T) {
+		m := getCompleteMetadata()
+		m.Properties["version"] = "not_valid_version"
+		meta, err := k.getKafkaMetadata(m)
+		assert.Error(t, err)
```
Comment (pkedy) [inline pubsub/kafka/kafka_test.go:78]: Nit: Pls use `require.Error`

File: pubsub/kafka/kafka_test.go [hunk 3]
```diff
@@ -40,16 +40,49 @@ func getBaseMetadata() pubsub.Metadata {
...
+
+func TestParseMetadata(t *testing.T) {
 	k := getKafkaPubsub()
-	meta, err := k.getKafkaMetadata(m)
-	assert.Nil(t, err)
+	t.Run("default kafka version", func(t *testing.T) {
+		m := getCompleteMetadata()
+		meta, err := k.getKafkaMetadata(m)
+		require.NoError(t, err)
+		assert.NotNil(t, meta)
```
Comment (pkedy) [inline pubsub/kafka/kafka_test.go:59]: Nit: Pls use `require.NotNil`

**Review by CodeMonkeyLeet — COMMENTED** [review]
LGTM, thanks for the contribution! 

Per the issue template, can you also open an issue in dapr/docs to track that the new `Version` metadata field is documented in https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-apache-kafka/? It helps us make sure that new public options are documented in the same versions that the functionality is released in.

**Review by pkedy — APPROVED** [review]
This is great! I know users have asked for this. I approved but made comments on a few testing nitpicks.

File: pubsub/kafka/kafka_test.go (hunk 1 above)
Comment (krutsko) [reply pubsub/kafka/kafka_test.go:59]: there are 2 independent tests where need to `assert` each own `metadata`. Maybe I'm missing the ask here.

**Review by CodeMonkeyLeet — APPROVED** [review]

**Review by daixiang0 — APPROVED** [review]

**Review by mukundansundar — APPROVED** [review]

File: pubsub/kafka/kafka_test.go (hunk 1 above)
Comment (pkedy) [reply pubsub/kafka/kafka_test.go:59]: I think I misread the code. Disregard :)

**Review by pkedy — APPROVED** [review]

**Review by beiwei30 — APPROVED** [review]

**Review by artursouza — APPROVED** [review]

**Pull request merged**